### ISSUE TITLE
Inconsistent provisioning EC2 AMIs

### ISSUE THREAD

**Issue by erez-rabih** [mitchellh/packer#2033]
Hi,

I'm using packer to bake AWS AMIs using chef provisioning.

Sometimes, after the chef run is over and the image is actually ready packer fails on adding tags to the AMI resource on amazon. It looks like it tries to add the tags immediately after registering the AMI and Amazon does not recognise the AMI as a valid resource.
Again, sometimes it finishes successfully and other times it fails, nothing consistent.
This is an example output of the error I'm getting:

```
12:02:28 ==> amazon-ebs: Stopping the source instance...
12:02:28 ==> amazon-ebs: Waiting for the instance to stop...
12:02:53 ==> amazon-ebs: Creating the AMI: Base Image. Build 66486 2015-04-12T08-50-03Z
12:02:53     amazon-ebs: AMI: ami-51b09b61
12:02:53 ==> amazon-ebs: Adding tags to AMI (ami-51b09b61)...
12:02:53     amazon-ebs: Adding tag: "Type": "Base_Image"
12:02:53     amazon-ebs: Adding tag: "BuildNumber": "66486"
12:02:53     amazon-ebs: Adding tag: "By": "Jenkins"
12:02:53     amazon-ebs: Adding tag: "Created": "1428828603"
12:02:53 ==> amazon-ebs: Error adding tags to AMI (ami-51b09b61): The image id '[ami-51b09b61]' does not exist (InvalidAMIID.NotFound)
12:02:53 ==> amazon-ebs: Terminating the source AWS instance...
12:03:00 ==> amazon-ebs: Deleting temporary keypair...
12:03:00 Build 'amazon-ebs' errored: Error adding tags to AMI (ami-51b09b61): The image id '[ami-51b09b61]' does not exist (InvalidAMIID.NotFound)
```

Any way to make sure I don't get those errors?
I am using version 0.6.2 of packer.
Thanks in advance.

**Comment by henrysher** [comment]
This issue still existed in the latest packer after integrating with **aws/aws-sdk-go**. It is caused by AWS API which is eventually consistent. In a summary, these may encounter this kind of issue from my experience:
1. instance id (frequently, especially when i built a large number of AMIs in parallel)
2. image id
3. security group
4. ssh key pair 

according to [AWS' suggestion](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/query-api-troubleshooting.html#eventual-consistency), we'd better to tolerate the errors and check the status via **Describe** command after instance launched or image created, and also with an exponential backoff algorithm to start seconds of wait time, and increase gradually up to about five minutes.

**Comment by erez-rabih** [comment]
For me, updating packer to 0.7.5 resolved the issue.

**Comment by catsby** [comment]
@henrysher – have you experienced this on 0.7.5? 

Thanks for reporting back @erez-rabih , though @henrysher is still right we need to tolerate this better and add minimal polling.

**Comment by mitchellh** [comment]
This shouldn't happen anymore, I imagine its fixed in 0.7.5. We now wait before adding tags.
